File: vowel-detector/vowel-detector.cpp
#include <iostream>
#include <string>
#include <map>

int main() {
    std::string inputString;
    std::cout << "Enter a string: ";
    std::getline(std::cin, inputString);

    std::string vowels = "aeiouAEIOU";
    int totalVowels = 0;
    std::map<char, int> vowelCounts;

    for (char c : inputString) {
        if (vowels.find(c) != std::string::npos) {
            totalVowels++;
            vowelCounts[c]++;
        }
    }

    std::cout << "Original string: " << inputString << std::endl;
    std::cout << "Total number of vowels: " << totalVowels << std::endl;
    std::cout << "Vowel counts (case-sensitive):" << std::endl;
    for (auto const& [vowel, count] : vowelCounts) {
        std::cout << "  " << vowel << ": " << count << std::endl;
    }

    return 0;
}

File: vowel-detector/vowel-detector.py
input_string = input("Enter a string: ")

vowels = ['a', 'e', 'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U']
total_vowels = 0
vowel_counts = {}

for char in input_string:
    if char in vowels:
        total_vowels += 1
        vowel_counts[char] = vowel_counts.get(char, 0) + 1

print(f"Original string: {input_string}")
print(f"Total number of vowels: {total_vowels}")
print("Vowel counts (case-sensitive):")
for vowel, count in vowel_counts.items():
    print(f"  {vowel}: {count}")